cmd/dockerd/daemon.go

```diff
@@ -161,9 +161,7 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 		}
 	}
 
-	cli.api = apiserver.New(serverConfig)
```

Comment:
Where does the actual initialisation of the `Server` struct happen now? (i should probably check out the code locally 😂)

Also was scratching my head a bit why the "constructor" didn't actually construct everything, other than setting the "config" (but none of the other fields). Is that something we would "want" to be more of an actual constructor instead? Or not worth bothering?

Reply:
> Where does the actual initialisation of the `Server` struct happen ~now~?

Initialization of the `apiserver.Server` struct happens in the `cli.initMiddlewares` and `initRouter` calls. That is true on master and it is true in this PR branch. The only meaningful change is that `cli.api` is no longer a pointer.

> Also was scratching my head a bit why the "constructor" didn't actually construct everything, other than setting the "config" (but none of the other fields). Is that something we would "want" to be more of an actual constructor instead? Or not worth bothering?

I'm not convinced it's worth bothering. `apiserver.Server` implements the builder pattern, which seems to work well enough for this use case. I wish that its build method was named something more intuitive than `Wait()`, but that's out of scope for this PR.